tests/spread/plugins/v1/ruby/snaps/ruby-bundle-install/snap/snapcraft.yaml

```diff
@@ -16,6 +16,8 @@ parts:
     plugin: ruby
     source: .
     use-bundler: true
-    ruby-version: "2.6.10" # bundler requires ruby >= 2.6.0
+    ruby-version: "3.3.4" # bundler requires ruby >= 3.0.0
     build-environment:
       - BUNDLE_GEMFILE: "$SNAPCRAFT_PART_SRC/Gemfile"
+    build-packages:
```

Comment:
@cmatsuoka, should I add this as a build-package when `use-bundler: true` is defined?  

Another solution would be to add a new property `ruby-bundler-version` so projects can be pinned to an older version of the bundler that isn't dependent on `libyaml-dev`.

Or I could do nothing - snap maintainers would just have to add `build-package: [libyaml-dev]` to their project.

Reply:
Per a conversation with @cmatsuoka, we can:

* add a `bundler-version` property.  If it is not defined, then install the latest version
* check which version of the bundler required `libyaml-dev`.  If `bundler-version` is not set or is greater than that version, then snapcraft will include `libyaml-dev` as a build-package

When we port the ruby plugin to craft-parts, we can redesign `ruby-use-bundler` to accept a version string instead of a boolean flag.